lib/vagrant/util/directory_changed.rb

```diff
@@ -0,0 +1,23 @@
+require 'pathname'
+
+module Vagrant
+  module Util
+    class DirectoryChanged
```

Comment:
It would be nice to have this something a little more generic, like just `Directory`, so that other directory related utility methods could be included here as well.